### PR TITLE
feat(build): add sourcemap option

### PR DESCRIPTION
**Pull request by filipesilva**
Add `--no-sourcemap` option to `ng build/serve/test`.

Disabling sourcemaps can help with build speeds. My tests on a medium project show a 11.5% improvement on initial builds, and a 37% improvement on rebuilds.

Disabling sourcemaps will make debugging harder, since it makes line information very innacurate.

Partially address #1980

### CODE REVIEW

File: packages/angular-cli/commands/serve.ts [hunk 1]
```diff
@@ -26,7 +26,9 @@ export interface ServeTaskOptions {
   sslKey?: string;
   sslCert?: string;
   aot?: boolean;
+  sourcemap?: boolean;
   open?: boolean;
+
```
Comment (hansl) [inline packages/angular-cli/commands/serve.ts]: Revert this empty line, please.

**Comment by naveedahmed1** [comment]
Should disabling sourcemaps also reduce the bundle size? I am using lazy loading and with --prod and --aot options the total size of all bundles is ~14mb with one of the bundles (0.xxxxxxxx.bundle.js) having size of ~8mb. The build process takes 20 minutes.

**Comment by naveedahmed1** [comment]
@filipesilva I just tried no-sourcemap using below command ng build --aot --prod --no-sourcemap, the total bundles size which was 14mb with previous version, is now 38.3 mb though the build which previousl took 20 minutes is now very quick.

**Comment by grizzm0** [comment]
@filipesilva Isn't sourcemap only used for dev/debug purposes?

My main.bundle.js.gz went from 301kb to 720kb with --no-sourcemap. More than double in size.

**Comment by rerickson1** [comment]
I see an increase in my bundle size if I use the --no-sourcemap option as well.
With `ng build --prod --aot` main.bundle.js is 825 KB, main.bundle.js.gz is 173 KB.
With `ng build --prod --aot --no-sourcemap` main.bundle.js is 2,837 KB, main.bundle.js.gz is 448 KB.
That's a 3.43 or 2.58 times increase when using the --no-sourcemap option.  It looks like I would be better off using the --no-sourcemap option in production because my pages will load faster.

**Comment by filipesilva** [comment]
@naveedahmed1 @grizzm0 @rerickson1 my bad, fix in https://github.com/angular/angular-cli/pull/3262

**Comment by naveedahmed1** [comment]
@filipesilva I tried angular-cli: 1.0.0-beta.21 and with ng build --aot --prod --no-sourcemap I am still getting the total bundles size of 38.4Mb while if I dont use the --no-sourcemap, the size of the generated bundles is 14.3

**Comment by filipesilva** [comment]
@naveedahmed1 #3262 has been merged but will only be out in the next release.

**Pull request merged**